### ISSUE TITLE
Modularize CPU backend methods

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs#4094]
### List of methods on the backend.ts interface. 

These are methods that need to be modularized and removed from the backend class.

- [x] batchMatMul
- [x] fusedBatchMatMul
- [x] slice
- [x] stridedSlice
- [x] unstack
- [x] reverse
- [x] concat
- [x] neg
- [x] add
- [x] addN
- [x] subtract
- [x] multiply
- [x] realDivide
- [x] divNoNan
- [x] floorDiv
- [x] sum
- [x] prod
- [x] unsortedSegmentSum
- [x] argMin
- [x] argMax
- [x] equal
- [x] notEqual
- [x] less
- [x] lessEqual
- [x] greater
- [x] greaterEqual
- [x] logicalNot
- [x] logicalAnd
- [x] logicalOr
- [x] where
- [x] select
- [x] topk
- [x] min
- [x] minimum
- [x] mod
- [x] max
- [x] maximum
- [x] all
- [x] any
- [x] squaredDifference
- [x] ceil
- [x] floor
- [x] round
- [x] sign
- [x] isNaN
- [x] isInf
- [x] isFinite
- [x] pow
- [x] exp
- [x] expm1
- [x] softmax
- [x] log
- [x] log1p
- [x] sqrt
- [x] rsqrt
- [x] square
- [x] reciprocal
- [x] relu
- [x] relu6
- [x] prelu
- [x] elu
- [x] eluDer
- [x] selu
- [x] int
- [x] clip
- [x] abs
- [x] complexAbs
- [x] sigmoid
- [x] softplus
- [x] sin
- [x] cos
- [x] tan
- [x] asin
- [x] acos
- [x] atan
- [x] atan2
- [x] sinh
- [x] cosh
- [x] tanh
- [x] asinh
- [x] acosh
- [x] atanh
- [x] erf
- [x] step
- [x] fusedConv2d
- [x] conv2d
- [x] conv2dDerInput
- [x] conv2dDerFilter
- [x] fusedDepthwiseConv2D
- [x] depthwiseConv2D
- [x] depthwiseConv2DDerInput
- [x] depthwiseConv2DDerFilter
- [x] conv3d
- [x] conv3dDerInput
- [x] conv3dDerFilter
- [x] maxPool
- [x] maxPoolBackprop
- [x] avgPool
- [x] avgPoolBackprop
- [x] avgPool3d
- [x] avgPool3dBackprop
- [x] maxPool3d
- [x] maxPool3dBackprop
- [x] reshape
- [x] cast
- [x] tile
- [x] pad
- [x] transpose
- [x] gather
- [x] gatherND
- [x] scatterND
- [x] batchToSpaceND
- [x] spaceToBatchND
- [x] resizeBilinear
- [x] resizeBilinearBackprop
- [x] resizeNearestNeighbor
- [x] resizeNearestNeighborBackprop
- [x] batchNormalization
- [x] localResponseNormalization4D
- [x] LRNGrad
- [x] multinomial
- [x] oneHot
- [x] cumsum
- [x] nonMaxSuppression
- [x] fft
- [x] ifft
- [x] complex
- [x] real
- [x] imag
- [x] cropAndResize
- [x] depthToSpace
- [x] split
- [x] sparseToDense
- [x] diag
- [x] fill
- [x] onesLike
- [x] zerosLike
- [x] linspace
